src/views/forms/AccountManagementFormsNew/SlippageEditor.tsx

```diff
@@ -0,0 +1,150 @@
+import { useEffect, useRef, useState, type Dispatch, type SetStateAction } from 'react';
+
+import type { NumberFormatValues } from 'react-number-format';
+import styled from 'styled-components';
+
+import { ButtonShape, ButtonSize } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+
+import { FormInput } from '@/components/FormInput';
+import { Icon, IconName } from '@/components/Icon';
+import { InputType } from '@/components/Input';
+import { Output, OutputType } from '@/components/Output';
+import { ToggleGroup } from '@/components/ToggleGroup';
+import { WithConfirmationPopover } from '@/components/WithConfirmationPopover';
+
+enum EditorState {
+  Viewing = 'Viewing',
+  Selecting = 'Selecting',
+  Editing = 'Editing',
+}
+
+type ElementProps = {
+  slippage: number;
+  setIsEditing?: Dispatch<SetStateAction<boolean>>;
+  setSlippage: (slippage: number) => void;
+  disabled?: boolean;
+};
+
+export type SlippageEditorProps = ElementProps;
+
+export const SlippageEditor = ({
+  disabled,
+  slippage,
+  setIsEditing,
+  setSlippage,
+}: SlippageEditorProps) => {
+  const percentSlippage = slippage * 100;
+  const [slippageInputValue, setSlippageInputValue] = useState(percentSlippage.toString());
+  const [editorState, setEditorState] = useState(EditorState.Viewing);
+
+  const stringGetter = useStringGetter();
+
+  const slippageOptionItems = [
+    { label: '0.1%', value: '0.001' },
+    { label: '0.25%', value: '0.0025' },
+    { label: stringGetter({ key: STRING_KEYS.CUSTOM }), value: 'custom' },
+  ];
+  const toggleGroupRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setIsEditing?.(editorState !== EditorState.Viewing);
+
+    if (editorState === EditorState.Selecting) {
+      toggleGroupRef.current?.focus();
+    } else if (editorState === EditorState.Editing) {
+      inputRef.current?.focus();
+    }
+  }, [editorState, setIsEditing]);
+
+  const onOpenChange = (isOpen: boolean) => {
+    if (!isOpen) {
+      setEditorState(EditorState.Viewing);
+    }
+  };
+
+  const onCancel = () => {
+    setEditorState(EditorState.Viewing);
+    setSlippageInputValue(percentSlippage.toString());
+  };
+
+  const onChangeSlippage = ({ value }: NumberFormatValues) => {
+    setSlippageInputValue(value);
+  };
+
+  const onConfirmSlippage = () => {
+    setEditorState(EditorState.Viewing);
+    setSlippage(Number(slippageInputValue) / 100);
+  };
+
+  const onSelectSlippage = (newSlippage: string) => {
+    if (newSlippage === 'custom') {
+      setEditorState(EditorState.Editing);
+    } else {
+      setSlippage(Number(newSlippage));
```

Comment:
validate the newSlippage string here?

Reply:
this one probably doesn't need to be validated bc it's just selecting from the hardcoded values we wrote. we should be validating in `onConfirmSlippage` tho! will add that

Reply:
ah i see - would be nice to narrow the types for `newSlippage` then:

```
const onSelectSlippage = (newSlippage: 'custom' | '0.001' | '0.0025')
```
(mega nit)

Reply:
woof i missed this! nw i'll roll this into the next PR

Reply:
all good!